lib/backend/webgl/src/lib.rs

```diff
@@ -109,3 +111,45 @@ fn handle_error<T: CompilationTarget>(ctx:&Context, target:ErrorTarget, t:T) ->
         Err(error)
     }
 }
+
+
+// ========================
+// === Managing buffers ===
+// ========================
+
+/// Set the array buffer data with floats.
+pub fn set_buffer_data(gl_context:&Context, buffer:&WebGlBuffer, data:&[f32]) {
+    let target = Context::ARRAY_BUFFER;
+    gl_context.bind_buffer(target,Some(&buffer));
+    set_bound_buffer_data(gl_context,target,data);
+}
+
+fn set_bound_buffer_data(gl_context:&Context, target:u32, data:&[f32]) {
+    let usage      = Context::STATIC_DRAW;
+    unsafe { // Note [unsafe buffer_data]
+        let float_array = Float32Array::view(&data);
+        gl_context.buffer_data_with_array_buffer_view(target,&float_array,usage);
+    }
+}
+
+/// Set the array buffer fragment with with floats.
+pub fn set_buffer_subdata(gl_context:&Context, buffer:&WebGlBuffer, offset:usize, data:&[f32]) {
+    let target = Context::ARRAY_BUFFER;
+    gl_context.bind_buffer(target,Some(&buffer));
+    set_bound_buffer_subdata(gl_context,target,offset as i32,data);
+}
+
+fn set_bound_buffer_subdata(gl_context:&Context, target:u32, offset:i32, data:&[f32]) {
+    unsafe { // Note [unsafe buffer_data]
+        let float_array = Float32Array::view(&data);
+        gl_context.buffer_sub_data_with_i32_and_array_buffer_view(target,offset,&float_array);
+    }
+}
+
+/* Note [unsafe buffer_data]
+ *
+ * The Float32Array::view is safe as long there are no allocations done
+ * until it is destroyed. This way of creating buffers were taken from
+ * wasm-bindgen examples
+ * (https://rustwasm.github.io/wasm-bindgen/examples/webgl.html)
+ */
```

Comment:
I would propose to always keep safety doc on the function which is a wrapper for some unsafe functionality. Jumping to a definition by labels makes the code harder to navigate. Do you agree?

Reply:
Btw, you can reuse here our existing functionality. Please see `buffer/item.rs` - it implements the same unsafe buffer view for a variety of types.